src/test/java/com/vinyl/AddProductToCartTest.java

```diff
@@ -0,0 +1,132 @@
+package com.vinyl;
+
+import com.vinyl.helper.TokenHeaderHelper;
+import com.vinyl.model.Cart;
+import com.vinyl.model.Product;
+import com.vinyl.model.ProductCart;
+import com.vinyl.model.Token;
+import com.vinyl.modelDTO.AddProductToCartDTO;
+import com.vinyl.repository.CartRepository;
+import com.vinyl.repository.ProductCartRepository;
+import com.vinyl.repository.TokenRepository;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.util.Lists;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
+
+import java.time.LocalDate;
+
+public class AddProductToCartTest extends BaseIntegration {
+
+    @Autowired
+    private TokenHeaderHelper tokenHeaderHelper;
+
+    @Autowired
+    private ProductCartRepository productCartRepository;
+
+    @Autowired
+    private CartRepository cartRepository;
+
+    @Autowired
+    private TokenRepository tokenRepository;
+
+    private AddProductToCartDTO request;
+
+    private Product product;
+
+    @Override
+    public void setUp() {
+        super.setUp();
+        product = defaultEntitiesHelper.createProduct();
+        request = new AddProductToCartDTO();
+        request.setQuantity(3);
+    }
+
+    public ResponseEntity<?> setUpHeaderAndGetTheResponse(Token token){
+        HttpHeaders headers=tokenHeaderHelper.setupToken(token.getHash());
+        ResponseEntity<?> response = trt.exchange("/products/" + product.getId()+ "/cart",HttpMethod.POST,new HttpEntity<>(request,headers),Void.class);
+
+        return response;
+    }
+
+    @Test
+    public void testWhenUserLoggedInAndItemExistsInCart(){
+        Token token = defaultEntitiesHelper.createToken(user);
+        Cart cart =defaultEntitiesHelper.createCart(user);
+        ProductCart pc=defaultEntitiesHelper.createProductCart(cart, product);
+        cart.setProducts(Lists.newArrayList(pc));
```

Comment:
also add a call to cartRepo.save after you add the product to the cart, please also do a debug session to see if this test executes as expected (when the endpoint is called the service will find an existing item in the cart).

Reply:
added the save call and debugged the test. everything executed as expected